### ISSUE TITLE
Try to use Web Workers for certain tasks to improve performance

### ISSUE THREAD

**Issue by stefandesu** [gbv/cocoda#632]
["Web Workers are a simple means for web content to run scripts in background threads."](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers) We could try to move certain tasks to a Web Worker in order to ease the JavaScript main thread and improve performance.

To start, we could perform loading concept data inside a Web Worker, for example.

[Vite natively supports Web Workers](https://vitejs.dev/guide/features.html#web-workers), so implementation should be easy. However, I'm not sure if we can use dependencies inside of Web Workers, so this might make things more difficult.

**Comment by stefandesu** [comment]
Some more resources on Web Workers:
- https://stenciljs.com/docs/web-workers
- https://web.dev/module-workers/
- https://leanylabs.com/blog/webworkers-intro/
- http://www.siddharthpandey.net/use-web-workers-speed-web-applications/

**Comment by nichtich** [comment]
> To start, we could perform loading concept data inside a Web Worker, for example.

I think cost of loading is mainly network but Web Workers can only help to speed up computation.

**Comment by stefandesu** [comment]
> I think cost of loading is mainly network but Web Workers can only help to speed up computation.

Yeah, that might be the case. As I'm currently finding out that Cocoda's slowness seems to come mainly from inefficient computed properties, this might not be necessary.

**Comment by stefandesu** [comment]
This might still be interesting to consider for certain cases, but it's not immediately relevant anymore, so I moved it to 2.0.0.
